calblueprint/ogsc
# WIP: Prisma 2.13 upgrade, profile field updates

* Upgrades the project to Prisma 2.13.1 (further upgrades are blocked by the need to bump the project to TS ^4.1.0).
  * Applies update-2.12 Prisma codemod to the codebase (changes manually applied to 3 files that caused jscodeshift to throw)
  * Removes copied enums from interfaces/user since prisma/prisma#3252 has been solved 🎉 
* Updates behavior of `updateEnumMigration` to remove existing rows with enum members that are being removed
* Adds migration for replacement of BMI profile field with height, team name with birth year
* Updates seeding script with new profile fields
* Adds new ProfileFieldValue for "distance measured," referring to a length value in inches

## TODO

* Resolve type errors that are occurring as a result of the profile field updates in components/Player/*
  * Validate that future profile field changes will not cause similar type errors to these components

## How to Test/Use PR feature

* Run migrations. Validate that no errors occur as a result of running migrations.
* Run seeding script. Validate that no errors occur as a result of running the seeding script.
* [wip] Verify that type errors are not surfacing as a result of the Prisma schema/type changes made by this PR.

## PR Checklist

Does your branch (check if true):

## Related PRs

Other PRs may conflict on the code updates that were required for bumping to Prisma 2.12+. I will manage these conflicts and ensure that this PR merges in cleanly.

## Migrations

* 20210216055344-UpdateProfileFields: Add new profile field updates, described above

CC: @sydbui
